src/betamax/recorder.py

```diff
@@ -166,3 +173,13 @@ def use_cassette(self, cassette_name, **kwargs):
             raise ValueError('Cassette must have a valid name and may not be'
                              ' None.')
         return self
+
+
+Configuration._Betamax_class = Betamax  # prevent circular import
+
+
+def dispatch_hooks(hook_name, *args):
```

Comment:
We could alter this to have the signature:

```py
def dispatch_hooks(hook_name, config, *args):
```

Such that it then looks like

```py
hooks = config.recording_hooks[hook_name]
for hook in hooks:
    hook(*args)
```

Reply:
I think that the hooks should be a class field on `Configuration` rather than an instance field on any given instance of the class. This is because calling `Betamax.configure()` returns a new instance of `Configuration()`. Does this work?

Reply:
`self` also has an instance of `Configuration` which won't be changed in the event someone has multiple threads that alter things. Further, since `config` is a public attribute a user /could/ alter it to add extra hooks if desired. In that case, we'll want to observe that by using the copy which won't affect the class vars.

Reply:
@sigmavirus24 I'm confused. 
I understand your suggestion to mean storing hooks on individual `Configuration` instances (initialized in `__init__`) rather than on the class as I currently have, but I believe this would have the effect of breaking [this configuration method](https://gist.github.com/jarhill0/72dd9e43d29f2b47d846edc263a0a7f2#file-hook_demo-py-L11-L13):

```python
with recorder.configure() as config:
    config.after_start(callback=before)
    config.before_stop(callback=after)
```

the `.configure()` method [returns a new `Configuration` instance](https://github.com/betamaxpy/betamax/blob/4009d74d0642fbc42a49ef2f27d16387711192e4/src/betamax/recorder.py#L81-L92) which is discarded when the `with` block ends, so the new hooks would not persist.

Reply:
This is a large part of why I detest what I did with this code oh so long ago :/

We can skip updating this for now.